tsl/src/hypercore/hypercore_handler.c

```diff
@@ -560,7 +569,11 @@ hypercore_rescan(TableScanDesc sscan, ScanKey key, bool set_params, bool allow_s
 	scan->reset = true;
 	scan->hs_scan_state = HYPERCORE_SCAN_START;
 
-	table_rescan(scan->cscan_desc, key);
+	if (key && key->sk_flags & SK_NO_COMPRESSED)
+		scan->hs_scan_state = HYPERCORE_SCAN_NON_COMPRESSED;
```

Comment:
I added a C-test for this. It is hard to test in a query since it requires engineering, e.g., a nested loop scan than changes the scan keys.

Also, if we'd like to test the rescan of only non-compressed data, this is an internal thing so needs  C test.